make/command

```diff
@@ -7,7 +7,15 @@ bin/cmdstan/%.o : src/cmdstan/%.cpp
 bin/print$(EXE) bin/stansummary$(EXE) bin/diagnose$(EXE) : CPPFLAGS_MPI =
 bin/print$(EXE) bin/stansummary$(EXE) bin/diagnose$(EXE) : LDFLAGS_MPI =
 bin/print$(EXE) bin/stansummary$(EXE) bin/diagnose$(EXE) : LDLIBS_MPI =
-bin/print$(EXE) bin/stansummary$(EXE) bin/diagnose$(EXE) : bin/%$(EXE) : bin/cmdstan/%.o
+bin/print$(EXE) bin/stansummary$(EXE) bin/diagnose$(EXE) : bin/%$(EXE) : bin/cmdstan/%.o $(BOOST)/stage/lib/libboost_program_options.a
 	@mkdir -p $(dir $@)
-	$(LINK.cpp) $^ $(LDLIBS) $(OUTPUT_OPTION)
+	$(LINK.cpp) $^ $(BOOST)/stage/lib/libboost_program_options.a $(LDLIBS) $(OUTPUT_OPTION)
```

Comment:
We want to link statically so we give the full path to the .a file here. On Unix there is no user-facing difference, on Windows this saves us from having to add stuff to PATH.